src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -10171,6 +10171,297 @@ partial class B
+        {
+            string source = $$"""
+                using System;
+                class C
+                {
+                    public object P1 => $"P1: {field is null}";
+                    public object P2 { get; set { field = value; field = $"{field}"; } }
+                }
+                class Program
+                {
+                    static void Main()
+                    {
+                        var c = new C();
+                        c.P2 = 2;
+                        Console.WriteLine(c.P1);
+                        Console.WriteLine(c.P2);
+                    }
+                }
+                """;
+            var comp = CreateCompilation(
+                source,
+                parseOptions: TestOptions.Regular.WithLanguageVersion(languageVersion),
+                options: TestOptions.ReleaseExe,
+                targetFramework: TargetFramework.Net80);
+
+            if (languageVersion == LanguageVersion.CSharp13)
+            {
+                comp.VerifyEmitDiagnostics(
+                    // (4,32): error CS0103: The name 'field' does not exist in the current context
+                    //     public object P1 => $"P1: {field is null}";
+                    Diagnostic(ErrorCode.ERR_NameNotInContext, "field").WithArguments("field").WithLocation(4, 32),
+                    // (5,19): error CS8652: The feature 'field keyword' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                    //     public object P2 { get; set { field = value; field = $"{field}"; } }
+                    Diagnostic(ErrorCode.ERR_FeatureInPreview, "P2").WithArguments("field keyword").WithLocation(5, 19),
+                    // (5,35): error CS0103: The name 'field' does not exist in the current context
+                    //     public object P2 { get; set { field = value; field = $"{field}"; } }
+                    Diagnostic(ErrorCode.ERR_NameNotInContext, "field").WithArguments("field").WithLocation(5, 35),
+                    // (5,50): error CS0103: The name 'field' does not exist in the current context
+                    //     public object P2 { get; set { field = value; field = $"{field}"; } }
+                    Diagnostic(ErrorCode.ERR_NameNotInContext, "field").WithArguments("field").WithLocation(5, 50),
+                    // (5,61): error CS0103: The name 'field' does not exist in the current context
+                    //     public object P2 { get; set { field = value; field = $"{field}"; } }
+                    Diagnostic(ErrorCode.ERR_NameNotInContext, "field").WithArguments("field").WithLocation(5, 61));
+            }
+            else
+            {
+                var verifier = CompileAndVerify(comp, verify: Verification.Skipped, expectedOutput: IncludeExpectedOutput("""
+                    P1: True
+                    2
+                    """));
```

Comment:
Consider calling `verifier.VerifyDiagnostics`. Applies to the tests below as well.